packages/shared/lib/services/account.service.ts

```diff
@@ -33,6 +33,20 @@ class AccountService {
         }
     }
 
+    async getAccountByUUID(uuid: string): Promise<Account | null> {
+        try {
+            const result = await db.knex.select('*').from<Account>(`_nango_accounts`).where({ uuid }).first();
```

Comment:
Just my personal opinion, maybe @TBonnin has some thoughts too:
That would be great for us to start having model functions that are a bit more pure (= less side effects). The problem with early try/catch it needs a lot of boilerplate to handle and does not actually make the error reporting much smoother since we never know if everything was try/catch correctly. 
It makes the code looks like go (lengthy functions with a lot of `if err != nil`) and it's especially annoying in tests imo
 
One good thing, it allows more precise error reporting (vs a global try/catch that needs more fine-tuning) but we could also just re-throw the error wrapped.

Reply:
tbh I think the `Result<>` we put in place has the same drawbacks I mentioned earlier. I think I just find that `throw new Error` is the most simple and effective way to handle this. At least for model functions. 

We don't use a lot of transaction but in our code base we already have a lot of intermediate state that needs to be handled at each `if (error)` (e.g: activity logs) and that's painful to handle without a try/catch.